zebra/zebra_vty.c

```diff
@@ -58,6 +58,7 @@
 #include "northbound_cli.h"
 #include "zebra/zebra_nb.h"
 #include "zebra/kernel_netlink.h"
+#include "bgpd/bgp_zebra.h"
```

Comment:
hmm - I'm not super crazy about pulling a bunch of bgp info into zebra - this would be the first instance of that, wouldn't it? maybe this opaque struct could go into zclient.h instead?

Reply:
I think this can be moved for sure.

Reply:
I do not like this being in zclient.h.  It's too much of a dumping ground already.  I would like a place that daemons could describe their data that they are going to be passing to zebra for opaqueness.  I also see it as a future thing for some other daemon using that data.  As an example.  In cisco ios you can pass community information from bgp to eigrp as opaque data that then can be used on the other side.

Reply:
So are you suggesting a new 'lib/daemon_opaque.h' file, something like that - a common place where these sorts of structs could be defined?

Reply:
So gentlemens, which way to proceed? 😀

Reply:
I'd move to `lib`, but if `zclient.h` is off limit I'd put into `lib/route_opaque.h`.

If the daemon/protocol uses some special/complex format (which I think is not the case here), then that header would contain only the necessary fields for the opaque data (to avoid unnecessary memory usage). I assume this is the way it is because `zebra` can't ask `bgpd` for this information on demand, right?

Reply:
@mjstapp @rzalamena which path do you recommend? `lib/route_opaque.h` for those per-daemon structs?